tests/unit-tests/test_inferrer.py

```diff
@@ -391,7 +480,1004 @@ def check_eval(mock_do_mini_batch_inference,
+        args = Namespace(yaml_config_file=os.path.join(Path(tmpdirname), 'gnn_nc.yaml'),
+                            local_rank=0)
+        config = GSConfig(args)
+
+        model5 = create_builtin_node_gnn_model(gdata.g, config, True)
+        inferrer5 = GSgnnNodePredictionInferrer(model5)
+        inferrer5.setup_device(device)
+
+        infer_dataloader5 = GSgnnNodeDataLoader(
+            gdata,
+            target_idx=gdata.get_node_test_set(config.target_ntype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=None,
+            train_task=False)
+
+        with assert_raises(AssertionError):
+            inferrer5.infer(
+                loader=infer_dataloader5,
+                save_embed_path='/tmp/embs',
+                use_mini_batch_infer=True
+                )
+
+    # delete temporary results
+    if os.path.exists('/tmp/embs'):
+        shutil.rmtree('/tmp/embs')
+
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+def test_rgcn_infer_ec4ef():
+    """ Test RGCN model Edge Classification inference pipeline with/without edge features.
+    """
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+
+    setup_device(0)
+    device = get_device()
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        gdata = GSgnnData(part_config=part_config)
+
+        # Test case 0: normal case, set RGCN model with edge features for EC, and provide
+        #              edge features.
+        #              Should complete inference process
+        create_config4ef(Path(tmpdirname), 'gnn_nc.yaml', task='ec', use_ef=True)
+        args = Namespace(yaml_config_file=os.path.join(Path(tmpdirname), 'gnn_nc.yaml'),
+                            local_rank=0)
+        config = GSConfig(args)
+
+        model1 = create_builtin_edge_gnn_model(gdata.g, config, True)
+        inferrer1 = GSgnnEdgePredictionInferrer(model1)
+        inferrer1.setup_device(device)
+
+        infer_dataloader1 = GSgnnEdgeDataLoader(
+            gdata,
+            target_idx=gdata.get_edge_test_set(config.target_etype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=config.edge_feat_name,
+            remove_target_edge_type=config.remove_target_edge_type,
+            train_task=False)
+ 
+        inferrer1.infer(
+            loader=infer_dataloader1,
+            save_embed_path=None,
```

Comment:
ditto
Can we check save_prediction_path?

Reply:
Yes, add the check of predictions